### ISSUE TITLE
Source postgres: Validate SSH key before checking connection.

### ISSUE THREAD

**Issue by alafanechere** [airbytehq/airbyte#10004]
If an invalid SSH key (not RSA format, typo during copy paste etc) is used the failure happens at connection check:
`Cannot invoke "org.bouncycastle.openssl.PEMKeyPair.getPublicKeyInfo()" because "keypair" is null` 

It would be great if we could validate the SSH key in the front-end with regex patterns.

**Comment by bleonard** [comment]
We should validate this as part of the `CHECK` command

**Comment by akashkulk** [comment]
@ryankfu is this a duplicate that will be resolved by your recent PR? https://github.com/airbytehq/airbyte/pull/18041/

**Comment by ryankfu** [comment]
@akashkulk yes, this appears to have been fixed with the recent PR #18041 at least based on the changes from Rodi's [PR](https://github.com/airbytehq/airbyte/pull/15833/files) which changes the retrieval method from `getPublicKeyInfo()` to `keyPairs.iterator().next()`

I will close this ticket as completed and update #18041 to include a description with this comment
